Foundation/NSJSONSerialization.swift

```diff
@@ -93,7 +93,74 @@ public class NSJSONSerialization : NSObject {
     /* Generate JSON data from a Foundation object. If the object will not produce valid JSON then an exception will be thrown. Setting the NSJSONWritingPrettyPrinted option will generate JSON with whitespace designed to make the output more readable. If that option is not set, the most compact possible JSON will be generated. If an error occurs, the error parameter will be set and the return value will be nil. The resulting data is a encoded in UTF-8.
      */
     public class func dataWithJSONObject(obj: AnyObject, options opt: NSJSONWritingOptions) throws -> NSData {
-        NSUnimplemented()
+        let newIndent = opt.contains(NSJSONWritingOptions.PrettyPrinted) ? "\t" : ""
+        let newLine = opt.contains(NSJSONWritingOptions.PrettyPrinted) ? "\n" : ""
+        let spacing = opt.contains(NSJSONWritingOptions.PrettyPrinted) ? " " : ""
+        let str = NSString(try JSONSerialize(obj, newIndent: newIndent, newLine: newLine, space: spacing))
+        return str.dataUsingEncoding(NSUTF8StringEncoding)!
+    }
+    
+    private class func JSONSerialize(object: AnyObject, newIndent: String, newLine: String, space: String, indentation: String = "") throws -> String
+    {
+        if let array = object as? [AnyObject]
+        {
+            guard array.count > 0
+                else
+            {
+                return "\(indentation)[]"
+            }
+            return try array.reduce("\(indentation)[\(newLine)", combine: {
+                let separator = ($1 === array.last) ? "\(newLine)\(indentation)]" : ",\(newLine)"
+                return $0 + (try JSONSerialize($1, newIndent: newIndent, newLine: newLine, space: space, indentation: indentation + newIndent)) + separator
+            })
+        }
+        else if let dict = object as? [String : AnyObject]
+        {
+            guard dict.count > 0
+                else
+            {
+                return "\(indentation){}"
+            }
+            var index = 0
+            return try dict.reduce("\(indentation){\(newLine)", combine: {
+                let valueString : String
+                if let str = $1.1 as? String
+                {
+                    let string = NSString(NSString(str).stringByReplacingOccurrencesOfString("\\", withString: "\\\\")).stringByReplacingOccurrencesOfString("\"", withString: "\\\"")
+                    // TODO: Escape control characters in string as required by RFC
+                    valueString = "\"\(string)\""
+                }
+                // TODO: Use NSNumber instead of NSDecimalNumber once implemented properly.
```

Comment:
NSDecimalNumber is not implemented, NSNumber is but the JSON types should be encoded as Double currently